client/src/pages/ContestSubmissions.js

```diff
@@ -0,0 +1,233 @@
+import React, { useState, useEffect } from 'react'
+
+import {
+    Typography,
+    Container,
+    Box,
+    Grid,
+    GridList,
+    GridListTile,
+    makeStyles,
+    Avatar,
+    Button,
+    Tab,
+    Tabs,
+    useMediaQuery,
+    Paper
+} from '@material-ui/core'
+import { useAuth } from "../components/UserContext";
+import { Link } from 'react-router-dom'
+
+export default function ContestSubmissions(props) {
+    const { authTokens } = useAuth();
+    const [user] = useState(authTokens ? authTokens.user : null);
+
+    const isMobile = useMediaQuery(theme => theme.breakpoints.down('xs'));
+    const [submissions, setSubmissions] = useState(imageGridList)
+    const [namedSubmissions, setNamedSubmissions] = useState(false)
+    const classes = useStyles()
+    const [activeTab, setActiveTab] = useState(0)
+    const contestId = props.match.params.id
+    const [contestInfo, setContestInfo] = useState({
+        creation_date: "",
+        deadline_date: "",
+        description: "",
+        prize_amount: 0,
+        title: "",
+        user_id: "",
+        _id: contestId
+    })
+    useEffect(() => {
+        const getInfo = async () => {
+            const contestInfo = await fetch(`/api/contest/${contestId}`, {
+                headers: {
+                    "Content-Type": "application/json",
+                    "x-auth-token": authTokens.token
+                },
+            })
+            let jsonContestInfo = await contestInfo.json()
+            setContestInfo(jsonContestInfo.contest)
+            setSubmissions(jsonContestInfo.submissions)
+
+        }
+        getInfo()
+    }, [user])
+    useEffect(() => {
+        const getNames = async () => {
```

Comment:
Instead of adding the names to each submission, we probably want to configure the BE /api/contest/:id controller instead, that should return an array of submissions with the user name appended

Reply:
Thanks. I've updated submissions to include names from the backend.